examples/Tests/ParametricEq.hpp

```diff
@@ -0,0 +1,585 @@
+    inputs.band2.id = 1;
+    inputs.band3.id = 2;
+    inputs.band4.id = 3;
+    inputs.band5.id = 4;
+    inputs.band6.id = 5;
+    inputs.band7.id = 6;
+    inputs.band8.id = 7;
+    std::cout << "initializing filters with pointers" << std::endl;
+    lowpassFilter = new Dsp::FilterDesign<Dsp::Butterworth::Design::LowPass<maxOrder>, chans>; // crashes when cutoffFreq = 0 (should never happen)
+    highpassFilter = new Dsp::FilterDesign<Dsp::Butterworth::Design::HighPass<maxOrder>, chans>;
+    bandpassFilter = new Dsp::FilterDesign<Dsp::Butterworth::Design::BandPass<maxOrder>, chans>;
+    bandstopFilter = new Dsp::FilterDesign<Dsp::Butterworth::Design::BandStop<maxOrder>, chans>;
+    lowshelfFilter = new Dsp::FilterDesign<Dsp::Butterworth::Design::LowShelf<maxOrder>, chans>;
+    highshelfFilter = new Dsp::FilterDesign<Dsp::Butterworth::Design::HighShelf<maxOrder>, chans>;
+    bandshelfFilter = new Dsp::FilterDesign<Dsp::ChebyshevI::Design::BandShelf<maxOrder>, chans>; // crashes when gain = 0;
+
+    Dsp::Params params; 
+
+    // This feels kinda messy because each filter type has its own parameters order;
+    // the two first parameters are always sample rate and filter order;
+    // for the other parameters, you need to check the TypeBase[I-IV] class of the desired filter (Butterworth, Chebyshev, etc).
+    params[0] = s.rate;   // sample rate
+    params[1] = 1;        // filter order
+    params[2] = 10.;     // cutoff frequency / center frequency
+
+    lowpassFilter->setParams(params);
+    highpassFilter->setParams(params);
+
+    params[3] = .5;     // frequency band width
+
+    bandpassFilter->setParams(params);
+    bandstopFilter->setParams(params);
+
+    params[3] = -6.;     // shelf gain
+
+    lowshelfFilter->setParams(params);
+    highshelfFilter->setParams(params);
+
+    params[3] = .5;     // frequency band width
+    params[4] = -6.;      // shelf gain
+    params[5] = .5;       // ripple
+
+    bandshelfFilter->setParams(params);
+
+    std::cout << "prepared" << std::endl;
+  }
+
+  void applyBand(
+      int N,
+      EQBand& bd,
+      filterToggle& tog,
+      order<maxOrder>& ord,
+      cutoffFrequency& cutoffFreq,
+      centerFrequency& centerFreq,
+      gain& gn,
+      frequencyBandWidth& freqBdWidth,
+      passbandRipple& ripple)
+  {
+    
+    if (tog.value)
+    {
+    // Fortunately one would say, here we don't have to remember the parameters order as the DspFilters lib provides ParamID.
+    // BUT access with ID is non-constant (see the implementation for yourself) while setParam isConstant. So for efficiency, we WILl be using parameters index rather than id.
+      auto rate = inputs.audio.sampleRate;
+      auto& out = outputs.audio;
+      Dsp::Params previousParams;
+      switch (bd.value)
+      {
+        case filters::lowpass:
+          previousParams = lowpassFilter->getParams();
+          lowpassFilter->setParam(1, ord.value);
+          lowpassFilter->setParam(2, cutoffFreq.value);
```

Comment:
i'm curious, there isn't a kCutoff enum or something ?

Reply:
The DSPFilters parameters order depends on the filter's implementation (e.g Butterworth, Chebyshev, etc); they do provide IDs for parameters and a setParamByID function but its complexity isn't constant (i.e it loops through all IDs till it finds the right one). I found this to be the cleanest way to work around (but unfortunately not the easiest to read)

Reply:
oh my, it's terrible that they did it like this aha, you did good !